### ISSUE TITLE
`cabal init` crashes if `git` is not installed

### ISSUE THREAD

**Issue by noiioiu** [haskell/cabal#10484]
**Describe the bug**
Running `cabal init` without `git` installed results in the following error:
```
git: readCreateProcessWithExitCode: execvp: does not exist (No such file or directory)
```

or

```
git: readCreateProcessWithExitCode: posix_spawnp: does not exist (No such file or directory)
```

**To Reproduce**
Run `cabal init` in an empty directory on a system without `git` installed or after setting `$PATH` to something that includes `cabal` and `ghc` but not `git`. Answer the prompts; the program will exit with the above error message after you choose a license.

**Expected behavior**
`cabal init` should continue running and ask for author name, email address, etc.

**System information**
 - `cabal` 3.10.3.0, `ghc` 9.6.5 on NixOs 24.05
 - `cabal` 3.12.1.0, `ghc` 9.6.6 on OpenBSD 7.6-current
 
**Additional context**
The function `Distribution.Client.Init.NonInteractive.Heuristics.guessGitInfo` uses `readProcessWithExitCode`, which throws an `IOError` which should be caught if the file is not found.

**Comment by geekosaur** [comment]
Huh. Supposedly this was fixed (https://github.com/haskell/cabal/issues/8478) unless I'm misreading.

**Comment by noiioiu** [comment]
No, it looks like the fix only deals with the case where `git` fails.

`readProcessWithExitCode` only returns an `ExitFailure` if `git` runs and exits unsuccessfully. If `git` is not found at all, it throws an exception.

**Comment by ffaf1** [comment]
@BasLaa will be able to diagnose quicker.

**Comment by noiioiu** [comment]
I have a [draft pull request](https://github.com/haskell/cabal/pull/10486) that _might_ fix this.

**Comment by ulysses4ever** [comment]
> Huh. Supposedly this was fixed (#8478) unless I'm misreading.

Haven't check in the code, but the catch maybe that #8478 talks about the non-interactive mode whereas this ticket is about interactive. Unfortunately, since at least the last init rewrite (it happened a couple of years ago), the two modes have very little code in common (in comparison to what one might expect).

**Comment by geekosaur** [comment]
I was misreading: that ticket fixed it when git was found but not configured. This one's about when git isn't found at all.

**Comment by ulysses4ever** [comment]
Did you read the description of #8478?

**Comment by ulysses4ever** [comment]
Maybe you're saying: #8755 didn't fix #8478 even though it claims that. That's sad but not surprising 'cause we didn't have a real integration test in #8755. We should do better this time around.
